rotkehlchen/tasks/assets.py

```diff
@@ -244,3 +260,74 @@ def update_owned_assets(user_db: DBHandler) -> None:
             (DBCacheStatic.LAST_OWNED_ASSETS_UPDATE.value, str(ts_now())),
         )
+
+
+def update_underlying_assets(chains_aggregator: 'ChainsAggregator') -> None:
+    """Fetch the underlying defi assets and polulate `underlying_tokens_list` in globaldb"""
+    for chain_id, data_provider_address in (
+        (ChainID.ETHEREUM, DATA_PROVIDER_ETH),
+        (ChainID.OPTIMISM, DATA_PROVIDER_EVM),
+        (ChainID.POLYGON_POS, DATA_PROVIDER_EVM),
+        (ChainID.ARBITRUM_ONE, DATA_PROVIDER_EVM),
+        (ChainID.BASE, DATA_PROVIDER_BASE),
+        (ChainID.GNOSIS, DATA_PROVIDER_GNO),
+        (ChainID.SCROLL, DATA_PROVIDER_SCRL),
+    ):
+        node_inquirer = chains_aggregator.get_evm_manager(chain_id=chain_id).node_inquirer  # type: ignore[arg-type]  # all iterated ChainIDs are supported
+        aave_data_provider = node_inquirer.contracts.contract(address=data_provider_address)
+
+        try:
+            underlying_tokens = [  # get all the underlying tokens in aave v3
+                to_checksum_address(token_address)
+                for _, token_address in aave_data_provider.call(
+                    node_inquirer=node_inquirer,
+                    method_name='getAllReservesTokens',
+                )
+            ]
+            reserve_tokens_addresses = node_inquirer.multicall(calls=[
+                (  # get all the reserve tokens for each underlying token
+                    aave_data_provider.address,
+                    aave_data_provider.encode(
+                        method_name='getReserveTokensAddresses',
+                        arguments=[underlying_token],
+                    ),
+                )
+                for underlying_token in underlying_tokens
+            ])
+        except RemoteError as e:
+            log.error(f'Failed to query Aave v3 reserve tokens addresses due to {e!s}')
+
+        # add the mappings of underlying tokens to the globaldb
+        for underlying_idx, underlying_token_address in enumerate(underlying_tokens):
+            for reserve_idx, decoded_reserve_token_address in enumerate(aave_data_provider.decode(
+                method_name='getReserveTokensAddresses',
+                arguments=[underlying_token_address],
+                result=reserve_tokens_addresses[underlying_idx],
+            )):  # we get three addresses per underlying token, first is the AToken and others are the Debt tokens  # noqa: E501
+                decoded_reserve_token = EvmToken.initialize(
+                    address=to_checksum_address(decoded_reserve_token_address),
+                    chain_id=chain_id,
+                    protocol=CPT_AAVE_V3,
+                    token_kind=EvmTokenKind.ERC20,
+                    underlying_tokens=[UnderlyingToken(
+                        address=underlying_token_address,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=ONE if reserve_idx == 0 else -ONE,  # 1 for ATokens, -1 for Debt Tokens  # noqa: E501
```

Comment:
This is a bad hack. The weights are values between 0 and 1. I guess this is to have negative prices but we don't support negative prices. What you should do is to move the debt tokens to the liabilities part of the balances instead of the assets. Add to the PR comment that is a step missing and that you are aware of it.